x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/steps_state_machine/steps_state_machine.ts

```diff
@@ -31,12 +32,20 @@ export const stepMachine = setup({
           resources?: ProcessorResources;
         }
       ) => {
+        const nextStep: StreamlangStepWithUIAttributes = {
+          ...params.step,
+          customIdentifier: context.step.customIdentifier,
+          parentId: context.step.parentId,
+        };
+
+        if (isActionBlock(context.step) && isActionBlock(nextStep)) {
+          if (context.step.description && !nextStep.description) {
+            nextStep.description = context.step.description;
+          }
+        }
```

Comment:
Could you please clarify which case this handles? Not sure I get it why we need to keep the previous description if the step that comes in `params` doesn't have it.